core/cold-wallet/wallet/cold-wallet.ts

```diff
@@ -0,0 +1,53 @@
+import { Network } from '@defichain/jellyfish-network';
+import { JellyfishWallet, WalletHdNode } from '@defichain/jellyfish-wallet';
+import { WhaleWalletAccount, WhaleWalletAccountProvider } from '@defichain/whale-api-wallet';
+import { Bip32Options, MnemonicHdNodeProvider } from '@defichain/jellyfish-wallet-mnemonic';
+import { WhaleApiClient } from '@defichain/whale-api-client';
+
+export class ColdWallet {
+  public static NEEDED_SEED_LENGTH = 24;
+  private seed: string[];
+  private network: Network;
+  private client: WhaleApiClient;
+
+  private wallet: JellyfishWallet<WhaleWalletAccount, WalletHdNode> | undefined;
+
+  constructor(seed: string[], network: Network) {
+    this.seed = seed;
+    this.network = network;
+    this.client = new WhaleApiClient({
+      url: process.env.OCEAN_URL,
+      version: process.env.OCEAN_VERSION,
+      network: network.name,
+    });
```

Comment:
Why do we have a whale client on a cold wallet?

Reply:
good point should have added, that this need to move out of cold-wallet, as soon as MN manager is able to do that :)

Will add a comment for this